File: Data Structures & Algorithms/first-missing-positive/submission-1.java
class Solution {
    public int firstMissingPositive(int[] nums) {
        int i=0;
        int res = -1;

        while(i < nums.length){  //1,2,4,5,6,3,1
            int idx = nums[i]-1;
            if(nums[i] > 0 && nums[i] <= nums.length && idx != i && nums[i] != nums[idx]){
                int temp = nums[i];
                nums[i] = nums[idx];
                nums[idx] = temp;
            }else{
                i++;
            }
        }


        for(i=0; i<nums.length; i++){
            if(nums[i] != i+1){
                res = i+1;
                break; 
            }
        }

        return res != -1 ? res : nums.length + 1;
    }
}

File: Data Structures & Algorithms/daily-temperatures/submission-1.py
class Solution:
    def dailyTemperatures(self, temp: List[int]) -> List[int]:
        
        n = len(temp)
        res = [0]*n
        stk = []

        for j in range(n-1, -1, -1):

            while len(stk) > 0 and temp[stk[-1]] <= temp[j]:
                stk.pop()
            
            if not stk:
                res[j] = 0
            else:
                res[j] = stk[-1] - j
            stk.append(j)   


        return res     




        